Automattic/woocommerce-payments
# Making table rows "clickable"

Fixes #484

#### Changes proposed in this Pull Request

The approach in this PR varies a bit from the description in #484. Instead of making table rows clickable, it only links the first column to the correct screen.

Reasons:

- Making the whole table row clickable would cause a lot of complications:
    - Table rows/cells cannot be wrapped by anchor elements (HTML rules). This would require the click to be accomplished through JavaScript instead, meaning that we'd manually have to manipulate the browser into understanding what's going on.
    - An accessibility fallback will be required.
    - Most importantly, it would be confusing to have additional links within clickable rows.
- This way we'd follow the pattern from both WordPress and WC-Admin.

cc @LevinMedia 

Changes:

- Rename the `DetailsLink` component to `DetailsIcon` for consistency.
- Create a new `DetailsLink` component to use for links within columns.
- Use the new component in all tables by making the first column clickable.
- Adjust snapshots.

#### Testing instructions

* Inspect each table (Transactions, Disputes, and Payouts) to make sure there is a linked **text** column.
* Verify that ℹ️ links are still clickable.
